### DISCUSSION TITLE
apple32Main and apple64Main implementation difference question

### DISCUSSION THREAD

**Discussion by weijunfeng** [russhwolf/multiplatform-settings#161] [Q&A]
I see some difference in implementation in apple32Mian and apple64Main sources, but I don't understand why

apple32Mian
<img width="1028" alt="image" src="https://github.com/russhwolf/multiplatform-settings/assets/18522198/1655470c-c4b1-40a5-8065-4dd07e3169bc">

apple64Main
<img width="995" alt="image" src="https://github.com/russhwolf/multiplatform-settings/assets/18522198/47caa3f6-179c-403f-9f31-f0b36897bdbc">

I see that ios depends on the api, and all the apis used are provided, and there is no difference in the platform.

can you tell me why? thanks

**Comment by russhwolf** [comment]
The `integerForKey()` and `setInteger()` functions on iOS use `NSInteger`, which changes width based on architecture. It's a Kotlin `Long` on 64-bit systems and an `Int` on 32-bit systems, so we need to convert to Kotlin's `Int` and `Long` differently depending on which system we're on. Furthermore, so we can't use `NSInteger` to save large `Long` values on 32-bit systems, so we use `String` under-the-hood instead.
